bentoml/_internal/bento/build_config.py

```diff
@@ -450,22 +525,28 @@ def write_to_bento(self, bento_fs: "FS", build_ctx: str) -> None:
 
         # write pip install args to a text file if applicable
         if pip_args:
-            with bento_fs.open(fs.path.join(py_folder, "pip_args.txt"), "w") as f:
+            with bento_fs.open(fs.path.combine(py_folder, "pip_args.txt"), "w") as f:
                 f.write(" ".join(pip_args))
 
-        if self.lock_packages:
+        if self.lock_packages:  # pragma: no cover
+            # This section is not covered by tests and we relies
+            # on jazzband/pip-tools test coverage
+
             # Note: "--allow-unsafe" is required for including setuptools in the
             # generated requirements.lock.txt file, and setuptool is required by
             # pyfilesystem2. Once pyfilesystem2 drop setuptools as dependency, we can
             # remove the "--allow-unsafe" flag here.
 
             # Note: "--generate-hashes" is purposefully not used here because it will
             # break if user includes PyPI package from version control system
+
+            # changelog: move from fs.path.join to fs.path.combine
+            # refers to https://docs.pyfilesystem.org/en/latest/reference/path.html#fs.path.combine
```

Comment:
remove this? we can document this in PR description

Reply:
This one doesn't seem resolved?